File: MS.NET/Applications/Desktop/WpfMvvmTest/BindingApp/VisitorModel.cs
using System;

namespace BindingApp
{
    public class Visitor
    {
        public string Name { get; set; }

        public int Frequency { get; set; } = 1;

        public DateTime Recent { get; set; } = DateTime.Now;

        public void Revisit()
        {
            Frequency += 1;
            Recent = DateTime.Now;
        }

    }

    public class VisitorModel
    {
        private Document<Visitor> dataStore;

        public VisitorModel()
        {
            dataStore = Document<Visitor>.Open("visitors.xml");
        }

        public Visitor[] ReadVisitors()
        {
            return dataStore.ToArray();
        }

        public void WriteVisitor(string name)
        {
            Visitor visitor = dataStore.Find(e => e.Name == name);

            if (visitor == null)
                dataStore.Add(new Visitor { Name = name });
            else
                visitor.Revisit();

            dataStore.Save();
        }
    }
}

File: Java/Applications/Desktop/SwingMVCTest/VisitorModel.java
package app;

public class VisitorModel extends javax.swing.table.AbstractTableModel{

	private Document<Visitor> store;

	public VisitorModel(){
		store = Document.open(Visitor.class, "visitors.xml");
	}

	@Override
	public int getColumnCount(){
		return 3;
	}

	@Override
	public String getColumnName(int i){
		String[] names = {"Name", "Frequency", "Recent"};
		return names[i];
	}

	@Override
	public int getRowCount(){
		return store.size();
	}

	@Override
	public Object getValueAt(int i, int j){
		Visitor visitor = store.get(i);
		Object[] row = {visitor.name, visitor.frequency, visitor.recent};
		return row[j];
	}

	public void register(String name){
		Visitor visitor = store.find(e -> e.name.equals(name));
		if(visitor == null)
			store.add(new Visitor(name));
		else
			visitor.revisit();
		store.save();
		fireTableDataChanged();
	}
}

